webapp/src/js/services/hydrate-messages.js

```diff
@@ -0,0 +1,60 @@
+var _ = require('underscore'),
+    lineageFactory = require('lineage');
+
+angular.module('inboxServices').factory('HydrateMessages',
+  function(
+    $q,
+    DB
+  ) {
+
+    'use strict';
+    'ngInject';
+
+    var lineage = lineageFactory($q, DB());
+
+    var hydrateMessage = function(docId, contactId, reportedDate) {
+      return lineage.fetchHydratedDoc(docId).then(function(doc) {
+        var contact = null, phone = null, message = null;
+        if(doc.kujua_message) {
+          for(var i = 0; i < doc.tasks.length; i++) {
+            var task = doc.tasks[i];
+            if(task.messages[0] && task.messages[0].contact._id === contactId) {
+              message = task.messages[0];
+              break;
+            }
+          }
+          contact = message.contact;
+          phone = message.to;
+        } else if (doc.sms_message) {
+          contact = doc.contact;
+          phone = doc.from;
```

Comment:
I think you need `message = doc.sms_message.message` here or the message will always be null.